### PR TITLE
[lilv] update to 0.24.24

### LINKED ISSUE #42530: [lilv] update to v0.24.24
### Library name

lilv

### New version number

0.24.24

### Other information that may be useful (release notes, etc...)

The VCPKG port is still on version 24.10 from Sep 27, 2020: https://github.com/lv2/lilv/tags

### PR DESCRIPTION
**Pull request by jimwang118**
Fixes https://github.com/microsoft/vcpkg/issues/42530
1. Update the following ports
   lilv 0.24.24
   lv2 1.18.10
  sord 0.16.16
  serd 0.32.2
2. add new port zix.
3. Modify all port compilation methods to use upstream meson compilation.
4. Repair lv2 dynamic library installation path.
5. Fixed the issue of zix function adaptation under Windows.

- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md).
- [x] SHA512s are updated for each updated download.
- [ ] ~The "supports" clause reflects platforms that may be fixed by this new version.~
- [ ] ~Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.~
- [ ] ~Any patches that are no longer applied are deleted from the port's directory.~
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is added to each modified port's versions file.

Usage test passed with x64-windows triplet.

### CODE REVIEW

**Comment by JoergAtGithub** [comment]
Please note, that there is now also another PR for the serd update: https://github.com/microsoft/vcpkg/pull/42539, as I created two seperated issues for updating serd and lilv.

File: ports/zix/vcpkg.json
Comment (WangWeiLin-MV) [inline ports/zix/vcpkg.json:1]: Port name checked: https://repology.org/project/zix/versions

File: versions/baseline.json
Comment (WangWeiLin-MV) [inline versions/baseline.json:1]: All `versions/*` checked with `vcpkg x-add-version`.

**Review by WangWeiLin-MV — APPROVED** [review]
The port installation tests pass with the following triplets:

* x64-windows

**Comment by JoergAtGithub** [comment]
Thank you for the big effort of updating all the sub-ports as well!

**Pull request merged**